### PR TITLE
Create a kustomize function to set the gateway on virtual services.     

### PR DESCRIPTION
**Pull request by jlewi**
* The gateway on our virtual services is misconfigured as described in
  kubeflow/gcp-blueprints#22 and kubeflow/manifests#1169

  * We should be using the istio gateway in the istio namespace and not
    the gateway in the kubeflow namespace

* To begin to fix that this pr adds a kustomize function to transform
  all the gateways in the virtual services. This makes it possible for
  configurations to start transitioning to the gateway in the istio namespace

 * We need this for GCP blueprints with ACM (kubeflow/gcp-blueprints#22)
   because we don't want to duplicate the gateway in that case.

* Clean up some comments in the existing kustomize functions.

### CODE REVIEW

**Comment by jlewi** [comment]
#388 should be merged first.

File: kustomize-fns/image-prefix/function.go [hunk 1]
```diff
@@ -99,8 +98,6 @@ func (f *ImagePrefixFunction) Filter(inputs []*yaml.RNode) ([]*yaml.RNode, error
 // replaceImage looks for an annotation changing docker image prefixes and if it is present
 // applies it to all the images.
 func (f *ImagePrefixFunction) replaceImage(r *yaml.RNode) error {
-	m , _  := r.GetMeta()
```
Comment (vpavlin) [inline kustomize-fns/image-prefix/function.go:102]: How is this change related to this PR?

File: kustomize-fns/virtual-service/function.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+var _ kio.Filter = &VirtualServiceFunction{}
+
+// Filter returns a new VirtualServiceFunction
+func Filter() kio.Filter {
+	return &VirtualServiceFunction{}
+}
+
+// VirtualServiceFunction implements the ImagePrefix Function
+type VirtualServiceFunction struct {
```
Comment (vpavlin) [inline kustomize-fns/virtual-service/function.go:26]: Would it make sense to use Istio client-go here instead of defining the structs? https://github.com/istio/client-g

Comment (jlewi) [reply kustomize-fns/virtual-service/function.go:26]: The structure isn't a virtual service or other defined ISTIO struct so I don't think we can use client go.

File: kustomize-fns/image-prefix/function.go (hunk 1 above)
Comment (jlewi) [reply kustomize-fns/image-prefix/function.go:102]: Miscellaneous cleanup. Not directly related.
